test/extended/dns/dns_libraries.go

```diff
@@ -0,0 +1,134 @@
+func checkForPodLogFailures(f *e2e.Framework, pod *kapiv1.Pod) {
+	By("submitting the pod to kubernetes")
+	podClient := f.ClientSet.CoreV1().Pods(f.Namespace.Name)
+	updated, err := podClient.Create(context.Background(), pod, metav1.CreateOptions{})
+	if err != nil {
+		e2e.Failf("Failed to create %s pod: %v", pod.Name, err)
+	}
+
+	w, err := f.ClientSet.CoreV1().Pods(f.Namespace.Name).Watch(context.Background(), metav1.SingleObject(metav1.ObjectMeta{Name: pod.Name, ResourceVersion: updated.ResourceVersion}))
+	if err != nil {
+		e2e.Failf("Failed to watch pods: %v", err)
+	}
+	ctx, cancel := context.WithTimeout(context.Background(), e2e.PodStartTimeout)
+	defer cancel()
+	if _, err = watchtools.UntilWithoutRetry(ctx, w, PodSucceeded); err != nil {
+		e2e.Failf("Failed: %v", err)
+	}
+
+	By("retrieving the pod logs")
+	r, err := podClient.GetLogs(pod.Name, &kapiv1.PodLogOptions{Container: "querier"}).Stream(context.Background())
+	if err != nil {
+		e2e.Failf("Failed to get pod logs %s: %v", pod.Name, err)
+	}
+
+	scan := bufio.NewScanner(r)
+	for scan.Scan() {
+		line := scan.Text()
+		if strings.Contains(line, "fail") {
+			e2e.Failf("DNS resolution failed: %s", line)
+		}
+	}
+}
+
+var _ = Describe("[sig-network-edge] DNS lookup", func() {
+	f := e2e.NewDefaultFramework("dns-libraries")
+	oc := exutil.NewCLI("dns-libraries")
+
+	// creates a simple Pod that is using the image under /testdata/dns, which performs a DNS query to make sure Go's DNS resolver works fine with OpenShift DNS.
+	It("using Go's DNS resolver", func() {
+		configClient, err := configclient.NewForConfig(f.ClientConfig())
+		if err != nil {
+			e2e.Failf("Failed to get config client: %v", err)
+		}
+		infra, err := configClient.ConfigV1().Infrastructures().Get(context.Background(), "cluster", metav1.GetOptions{})
+		if err != nil {
+			e2e.Failf("Failed to get cluster config: %v", err)
+		}
+		if infra.Status.PlatformStatus.Type == configv1.BareMetalPlatformType {
+			Skip("Skip DNS lookup using Go's DNS resolver on BareMetal as there is no external connection to pull the base image.")
+		}
+
+		dnsService, err := f.ClientSet.CoreV1().Services("openshift-dns").Get(context.Background(), "dns-default", metav1.GetOptions{})
+		if err != nil {
+			e2e.Failf("Failed to get default DNS service: %v", err)
+		}
+
+		goVersions := []string{"go-v1-16", "go-v1-17"}
+		for _, version := range goVersions {
+			By(fmt.Sprintf("creating build and deployment config for dns-libraries-%s", version))
+			err = oc.Run("create").Args("-f", exutil.FixturePath("testdata", "dns", version, "dns_libraries_go.yaml")).Execute()
+			if err != nil {
+				e2e.Failf("Failed to create build configs and deployment config for dns-libraries-go: %v", err)
+			}
+
+			By("starting the builder image build with a directory")
+			err = oc.Run("start-build").Args(fmt.Sprintf("dns-libraries-%s", version), fmt.Sprintf("--from-dir=%s", exutil.FixturePath("testdata", "dns", version))).Execute()
```

Comment:
Same comment here. I don't know how this can be done with this the kube clientset, but I'm curious if this could be a GO Object that is created via GO API directly (avoid "shelling out")